userspace/libsinsp/parsers.cpp

```diff
@@ -408,6 +408,12 @@ void sinsp_parser::process_event(sinsp_evt *evt)
 	case PPME_SYSCALL_DUP_X:
 		parse_dup_exit(evt);
 		break;
+	case PPME_SYSCALL_DUP2_X:
+		parse_dup_exit(evt);
+		break;
+	case PPME_SYSCALL_DUP3_X:
+		parse_dup_exit(evt);
```

Comment:
Maybe we can get also the flag information for `dup3` in the `sinsp_fdinfo_t`. WDYT about it?

Reply:
Not sure if it is useful. @FedeDP what do you think?

Reply:
I am not sure either; we are only setting the O_CLOEXEC flag right?
It seems that fdinfo has only these flags:
```
enum flags
	{
		FLAGS_NONE = 0,
		FLAGS_FROM_PROC = (1 << 0),
		//FLAGS_TRANSACTION = (1 << 1),
		FLAGS_ROLE_CLIENT = (1 << 2),
		FLAGS_ROLE_SERVER = (1 << 3),
		FLAGS_CLOSE_IN_PROGRESS = (1 << 4),
		FLAGS_CLOSE_CANCELED = (1 << 5),
		FLAGS_IS_SOCKET_PIPE = (1 << 6),
		FLAGS_IS_TRACER_FILE = (1 << 7),
		FLAGS_IS_TRACER_FD = (1 << 8),
		FLAGS_IS_NOT_TRACER_FD = (1 << 9),
		FLAGS_IN_BASELINE_R = (1 << 10),
		FLAGS_IN_BASELINE_RW = (1 << 11),
		FLAGS_IN_BASELINE_OTHER = (1 << 12),
		FLAGS_SOCKET_CONNECTED = (1 << 13),
		FLAGS_IS_CLONED = (1 << 14),
		FLAGS_CONNECTION_PENDING = (1 << 15),
		FLAGS_CONNECTION_FAILED = (1 << 16),
	};
```

@Andreagit97 am i right?

Reply:
This is right, but I was thinking about [`m_openflags` field](https://github.com/falcosecurity/libs/blob/78f67c4b64cd7ea3d407facfa169de953fd10750/userspace/libsinsp/fdinfo.h#L336) where we store information about open flags and `dup3` uses one of this open flags `O_CLOEXEC`. This is similar to what we do [here](https://github.com/falcosecurity/libs/blob/78f67c4b64cd7ea3d407facfa169de953fd10750/userspace/libsinsp/parsers.cpp#L2409). WDYT @alacuku @FedeDP?

Reply:
+1 from me, thanks Andre for spotting it!